### ISSUE TITLE
mask hyperparameters

### ISSUE THREAD

**Issue by xxchauncey** [yanghaha0908/FastHuBERT#5]
hi,

during pretraining, does it use specaugmentation before cnn downsampling and then also apply original hubert masking after cnn? if so, should the mask hyperparameters be decreased since original hubert's framerate is 20ms while fasthubert is 40ms?

**Comment by yanghaha0908** [comment]
During pretraining, we don't apply specaugmentation before cnn downsampling, but only apply original hubert masking after cnn. 
We don't change the mask hyperparameters. The main mask hyperparameters are *mask_prob* and *mask_len*. *mask_prob* doesn't need to be change. *mask_len* of 10 works better than 5, so mask_len is always 10 in our experiments, even if the frameshift becomes 40ms in fasthubert.

**Comment by yanghaha0908** [comment]
During pretraining, we don't apply specaugmentation before cnn downsampling, but only apply original hubert masking after cnn.
We don't change the mask hyperparameters. The main mask hyperparameters are mask_prob and mask_len. mask_prob doesn't need to be change. mask_len of 10 works better than 5, so mask_len is always 10 in our experiments, even if the frameshift becomes 40ms in fasthubert.

Sorry for the late reply.

----- 原始邮件 -----
发件人: "ChaunceyChan" ***@***.***>
收件人: "yanghaha0908/FastHuBERT" ***@***.***>
抄送: "Subscribed" ***@***.***>
发送时间: 星期一, 2024年 10 月 14日 下午 2:29:23
主题: [yanghaha0908/FastHuBERT] mask hyperparameters (Issue #5)

hi,

during pretraining, does it use specaugmentation before cnn downsampling and then also apply original hubert masking after cnn? if so, should the mask hyperparameters be decreased since original hubert's framerate is 20ms while fasthubert is 40ms?

-- 
Reply to this email directly or view it on GitHub:
https://github.com/yanghaha0908/FastHuBERT/issues/5

**Comment by yanghaha0908** [comment]
During pretraining, we don't apply specaugmentation before cnn downsampling, but only apply original hubert masking after cnn.
We don't change the mask hyperparameters. The main mask hyperparameters are mask_prob and mask_len. mask_prob doesn't need to be change. mask_len of 10 works better than 5, so mask_len is always 10 in our experiments, even if the frameshift becomes 40ms in fasthubert.
